src/OrchardCore.Modules/OrchardCore.Settings/Recipes/SettingsStep.cs

```diff
@@ -23,76 +23,109 @@ protected override async Task HandleAsync(RecipeExecutionContext context)
         var model = context.Step;
         var site = await _siteService.LoadSiteSettingsAsync();
 
+        if (model == null || site == null)
```

Comment:
I don't think `site` will ever be null here. I would just remove the second check instead I would add the following after this check:

```
site ??= new SiteSettings();
```